### PR TITLE
Don't lookup cache or db on empty API key

### PR DESCRIPTION
**Pull request by luluz66**

### CODE REVIEW

File: enterprise/server/auth/auth.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/buildbuddy-io/buildbuddy/server/util/log"
 	"github.com/buildbuddy-io/buildbuddy/server/util/lru"
 	"github.com/buildbuddy-io/buildbuddy/server/util/random"
-	"github.com/buildbuddy-io/buildbuddy/server/util/request_context"
+	requestcontext "github.com/buildbuddy-io/buildbuddy/server/util/request_context"
```
Comment (vadimberezniker) [inline enterprise/server/auth/auth.go:23]: This would be better saved for a separate PR.
Instead of introducing an alias here, it might be better to change the package name to match the filename.

File: enterprise/server/auth/auth.go [hunk 2]
```diff
@@ -568,6 +568,9 @@ func lookupUserFromSubID(env environment.Env, ctx context.Context, subID string)
 }
 
 func (a *OpenIDAuthenticator) lookupAPIKeyGroupFromAPIKey(ctx context.Context, apiKey string) (interfaces.APIKeyGroup, error) {
+	if apiKey == "" {
+		return nil, status.UnauthenticatedError("missing API keys")
```
Comment (vadimberezniker) [inline enterprise/server/auth/auth.go:572]: keys -> key

**Review by vadimberezniker — APPROVED** [review]

File: enterprise/server/auth/auth.go (hunk 1 above)
Comment (luluz66) [reply enterprise/server/auth/auth.go]: Thank you for catching this. vim somehow auto adds it. Reverted

File: enterprise/server/auth/auth.go (hunk 2 above)
Comment (luluz66) [reply enterprise/server/auth/auth.go]: Done
